main.cpp

```diff
@@ -104,6 +114,394 @@ good to go!
 
 #include <iostream>
 
+struct FloatType;
+struct DoubleType;
+struct IntType;
+
+struct FloatType
+{
+    float* value = nullptr;
+
+    FloatType(float _value)
+    {
+        value = new float();
+        *value = _value;
```

Comment:
Avoid initializing in the constructor.  
use the member initializer list instead
- [ ] revise